LoopKit/CarbKit/CachedCarbObject+CoreDataClass.swift

```diff
@@ -51,13 +51,9 @@ class CachedCarbObject: NSManagedObject {
         }
     }
 
-    override func willSave() {
-        if isInserted {
-            if !changedValues().keys.contains("anchorKey") {
-                setPrimitiveValue(managedObjectContext!.anchorKey ?? 0, forKey: "anchorKey")
-            }
-        }
-        super.willSave()
+    override func awakeFromInsert() {
+        super.awakeFromInsert()
+        setPrimitiveValue(managedObjectContext!.anchorKey!, forKey: "anchorKey")
```

Comment:
[nit] I'm a tiny bit concerned this may crash unwrapping these optional values.  Is there any downside to just guarding it like this?
```suggestion
        guard let anchorKey = managedObjectContext?.anchorKey else { return }
        setPrimitiveValue(anchorKey, forKey: "anchorKey")
```
?

Reply:
Crashing is good if messier problems (like storing, and acting on bad data) would happen later from ignoring this failed assumption.

Reply:
Agreed.